pkg/infoschema/infoschema_v2.go

```diff
@@ -795,33 +795,34 @@ func (is *infoschemaV2) SchemaSimpleTableInfos(ctx context.Context, schema model
 		return nil, nil // something wrong?
 	}
 
-retry:
-	dbInfo, ok := is.SchemaByName(schema)
-	if !ok {
+	// Ascend is much more difficult than Descend.
```

Comment:
what do you mean by `difficult`, difficult to implement?

Reply:
Handle it in Ascend order is more difficult than handle it in Descend order.

The data organize like this:

```
| t1                      | t2     | t3    |
| v5  v6  v7  v8 v10  v12 | v7 v13 | v2 v5 |
```

If we want infoschma snapshot at v10, the result should be `[t1.v10     t2.v7]`

In Descend order, the total test condition could be:

```
if is.infoSchema.schemaMetaVersion >= item.schemaVersion {
    if curr == nil || curr.tableName != tableItems[i].tableName {
          // add curr to the result
    }
}
```

But in Ascend order, you may have a try to see how much more difficult

 :-)